changelogs/fragments/34-env-var-changes.yml

```diff
@@ -0,0 +1,8 @@
+---
+breaking_changes:
+  - hashi_vault - ``url`` parameter's env var ``VAULT_ADDR`` now has lower precedence than INI value (https://github.com/ansible-collections/community.hashi_vault/issues/8).
+  - hashi_vault - ``auth_method`` parameter's env var ``VAULT_AUTH_METHOD`` has been replaced by ``ANSIBLE_HASHI_VAULT_AUTH_METHOD`` (https://github.com/ansible-collections/community.hashi_vault/issues/17).
+  - hashi_vault - ``token_file`` and ``token_path`` parameters' env vars ``VAULT_TOKEN_FILE`` and ``VAULT_TOKEN_PATH`` have been replaced by ``ANSIBLE_HASHI_VAULT_TOKEN_FILE`` and ``ANSIBLE_HASHI_VAULT_TOKEN_PATH`` (https://github.com/ansible-collections/community.hashi_vault/issues/15).
+  - hashi_vault - ``role_id`` and ``secret_id`` parameters' env vars ``VAULT_ROLE_ID`` and ``VAULT_SECRET_ID`` have been replaced with ``ANSIBLE_HASHI_VAULT_ROLE_ID`` and ``ANSIBLE_HASHI_VAULT_SECRET_ID`` (https://github.com/ansible-collections/community.hashi_vault/issues/20).
+minor_changes:
+  - hashi_vault - ``url`` can be specified via new ``ANSIBLE_HASHI_VAULT_ADDR`` env var (https://github.com/ansible-collections/community.hashi_vault/issues/8).
```

Comment:
@felixfontein opinions on formatting here? I could have done all removals in breaking, all additions in minor; or everything in breaking; could have added separate entries for the "paired" ones (`token_file` and `token_path`), etc.

For me, no clear winner among them, I figure you must have a preferred way for listing these :)

Reply:
Definitely using both sections is the best way. I think the way you split this up is good!

Reply:
Should I then change the others so that for example, `ANSIBLE_HASHI_VAULT_AUTH_METHOD` being added goes in `minor_changes`, and the removal is described in `breaking_changes`?

Reply:
Yes. It's probably good to have two entries, like `The xxx option has been depreacted and will be removed in 2.0.0, use yyy instead` in `deprecated_features`, and `The yyy option has been added` in `minor_changes`. `breaking_changes` is only needed for actually breaking changes, not for deprecations.